### PR TITLE
Upgrade OpenSSL to 3.0.2

### PR DESCRIPTION
**Pull request by nazar-pc**
Changes:
* Upgrade to 3.0.2
* riscv64 architecture support
* cleanups

### CODE REVIEW

**Comment by jpakkane** [comment]
The build failure is due to calling a shell script. Those are not available in the VS env. Is this a new script or did the old version also fail to build on Windows?

**Comment by nazar-pc** [comment]
This was never meant to pass Windows CI, failure with VisualStudio is by design

**Comment by nazar-pc** [comment]
Upgraded to 3.0.2 that has important bug fixes, would be great if someone can finally review this :slightly_smiling_face:

**Comment by jpakkane** [comment]
Looks fine by a cursory glance. However since this is a crypto lib, it would be nice if someone else would take a second look.

**Comment by jpakkane** [comment]
:crossed_swords: conflicts.

**Comment by jpakkane** [comment]
On this page it says "This branch cannot be rebased due to conflicts". Reloading does not help.

**Comment by nazar-pc** [comment]
Weird, I see this:
Знімок екрана з 2022-03-18 14-11-58

**Comment by jpakkane** [comment]
Please rebase anyway, because the web ui won't let me do it.

**Comment by nazar-pc** [comment]
Rebased. I see macOS CI was added, not sure if it will pass, might have to be ignored just like Windows, we'll see soon enough.

**Comment by eli-schwartz** [comment]
>     Makefile:6: *** This can be run only on Linux.  Stop.

I had been testing out the suppressions needed on macOS anyway.

**Comment by eli-schwartz** [comment]
Please rebase, that fix is already in git master. :)

File: .github/workflows/sanity_checks.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
       - name: Install packages
         run: |
           sudo apt-get update
-          sudo apt-get -y install build-essential python3-pip ninja-build nasm
+          sudo apt-get -y install build-essential python3-pip ninja-build nasm libtext-template-perl
```
Comment (eli-schwartz) [inline .github/workflows/sanity_checks.yml:17]: We should move this to debian_packages in ci_config.json

File: subprojects/packagefiles/openssl/meson.build [hunk 2]
```diff
@@ -47,26 +47,17 @@ c_args = []
...
 else
-  new_gas_or_nasm = (
-    find_program('as', required: false, version: '>=2.26').found() or
-    find_program('nasm', required: false, version: '>=2.11.8').found()
-  )
-  any_gas_or_nasm = (
-    new_gas_or_nasm or
+  gas_or_nasm = (
     find_program('as', required: false).found() or
     find_program('nasm', required: false).found()
```
Comment (eli-schwartz) [inline subprojects/packagefiles/openssl/meson.build:54]: Shouldn't we keep the minimum versions here?

File: subprojects/packagefiles/openssl/meson.build [hunk 3]
```diff
@@ -55,88 +55,124 @@ else
...
-  elif host_machine.cpu_family() == 'mips64' and host_machine.system() == 'linux'
-    arch_subdir = 'linux64-mips64'
+  if is_bsd
+    if is_x86
+      arch_subdir = 'BSD-x86'
+    elif is_x86_64
+      arch_subdir = 'BSD-x86_64'
+    else
+      asm = 'no-asm'
+    endif
```
Comment (eli-schwartz) [inline subprojects/packagefiles/openssl/meson.build:84]: In general, these `if os; if arch` style checks with repetitive else clauses seem like maybe they'd actually be less repetitive if we do `if os and arch` style checks... just simplifying the `host_machine` property comparison checks if you like.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: subprojects/packagefiles/openssl/meson.build (hunk 2 above)
Comment (nazar-pc) [reply subprojects/packagefiles/openssl/meson.build:54]: We could, but those versions are already so old, it is not possible that people would have modern enough Python to run Meson at the same time. So I didn't think it was worth polluting the code with additional version checks.

**Comment by jpakkane** [comment]
This now fails for some reason:

```
2022-03-21T09:56:09.3674440Z make[1]: Leaving directory '/home/runner/work/wrapdb/wrapdb/subprojects/packagefiles/openssl/node/deps/openssl/openssl'
2022-03-21T09:56:09.3950405Z Can't locate Text/Template.pm in @INC (you may need to install the Text::Template module) (@INC contains: /home/runner/work/wrapdb/wrapdb/subprojects/packagefiles/openssl/node/deps/openssl/config/../openssl/util/perl/OpenSSL ../openssl /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.30.0 /usr/local/share/perl/5.30.0 /usr/lib/x86_64-linux-gnu/perl5/5.30 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.30 /usr/share/perl/5.30 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at ./generate_headers.pl line 6.
2022-03-21T09:56:09.3951643Z BEGIN failed--compilation aborted at ./generate_headers.pl line 6.
2022-03-21T09:56:09.3954461Z make: Leaving directory '/home/runner/work/wrapdb/wrapdb/subprojects/packagefiles/openssl/node/deps/openssl/config'
2022-03-21T09:56:09.3957612Z make: *** [Makefile:72: generate_headers] Error 2
2022-03-21T09:56:09.3962287Z Preparing release for: openssl_3.0.2-1
```

[Full log here](https://pipelines.actions.githubusercontent.com/aYE2ZrmXq5L0OupCF5oTsikJo0OTK3poljn60R3WAS84UA9wBt/_apis/pipelines/1/runs/1426/signedlogcontent/2?urlExpires=2022-03-21T10%3A03%3A07.0698881Z&urlSigningMethod=HMACV1&urlSignature=zywwa35Tjz6Ov9sYuO5KLISOqZ4Vk9TYUqIz7Mg9zAs%3D).

**Comment by nazar-pc** [comment]
Above error means `libtext-template-perl` wasn't installed, despite I have added it to `debian_packages` as suggested. Looks like it is not accounted for in release CI job, fixing it now.

Above link was broken for me, here is link to failed CI job: https://github.com/mesonbuild/wrapdb/runs/5624833952?check_suite_focus=true

**Pull request merged**